test/basic_parser.cpp

```diff
@@ -481,7 +481,10 @@ class basic_parser_test
                 s.data(), s.size(),
                 ec);
             if(! BOOST_TEST(! ec))
+            {
+                ::test_suite::log_type() << "    failed to parse: " << s;
```

Comment:
Not a fan of this why didn't we keep it using the `log` data member?

Reply:
hmm it failed to compile for me for some reason. Will recheck

Reply:
the lambda needs to capture `this`, and functions which use the log or call functions (or lambdas) that use the log cannot be `static`